test/e2e/apiextensions/composition_test.go

```diff
@@ -0,0 +1,700 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiextensions
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/utils/pointer"
+
+	corev1 "k8s.io/api/core/v1"
+	kextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/crossplane/crossplane-runtime/pkg/resource/unstructured/composed"
+	extv1 "github.com/crossplane/crossplane/apis/apiextensions/v1"
+	v1 "github.com/crossplane/crossplane/apis/pkg/v1"
+)
+
+func TestCompositeResourceGetsReady(t *testing.T) {
+	cases := map[string]struct {
+		reason string
+		body   func() error
+	}{
+		"TestCompositeResourceGetsReady": {
+			reason: "Composite Resource should get ready.",
+			body: func() error {
+				ctx := context.Background()
+				s := runtime.NewScheme()
+				if err := v1.AddToScheme(s); err != nil {
+					return err
+				}
+				if err := extv1.AddToScheme(s); err != nil {
+					return err
+				}
+
+				cfg := ctrl.GetConfigOrDie()
+				c, err := client.New(cfg, client.Options{
+					Scheme: s,
+				})
+				if err != nil {
+					return err
+				}
+
+				// Dynamic client required to create nopresources
+				dc, err := dynamic.NewForConfig(cfg)
+				if err != nil {
+					return err
+				}
+
+				prv := &v1.Provider{
+					ObjectMeta: metav1.ObjectMeta{Name: "provider-nop"},
+					Spec: v1.ProviderSpec{
+						PackageSpec: v1.PackageSpec{
+							Package:                     "crossplane/provider-nop:main",
+							IgnoreCrossplaneConstraints: pointer.BoolPtr(true),
+						},
+					},
+				}
+
+				if err := c.Create(ctx, prv); err != nil {
+					t.Fatalf("Create provider %q: %v", prv.GetName(), err)
+				}
+
+				t.Logf("Created provider %q", prv.GetName())
+
+				t.Cleanup(func() {
+					t.Logf("Cleaning up provider %q.", prv.GetName())
+					if err := c.Get(ctx, types.NamespacedName{Name: prv.GetName()}, prv); resource.IgnoreNotFound(err) != nil {
+						t.Fatalf("Get provider %q: %v", prv.GetName(), err)
+					}
+					if err := c.Delete(ctx, prv); resource.IgnoreNotFound(err) != nil {
+						t.Fatalf("Delete provider %q: %v", prv.GetName(), err)
+					}
+					t.Logf("Deleted provider %q", prv.GetName())
```

Comment:
We could run into flakiness in this test because packages are blocked from deletion by their revisions which could cause installation of the provider in a subsequent test to fail if the previous hasn't been completely cleaned up yet, but in reality cleanup should be almost instantaneous, so I feel comfortable with this strategy and we can revisit if this is causing problems in our integration suite :+1: